beacon_node/network/src/sync/range_sync/batch.rs

```diff
@@ -139,8 +150,13 @@ impl<T: EthSpec, B: BatchConfig> BatchInfo<T, B> {
     /// Epoch boundary |                                   |
     ///  ... | 30 | 31 | 32 | 33 | 34 | ... | 61 | 62 | 63 | 64 | 65 |
     ///       Batch 1       |              Batch 2              |  Batch 3
-    pub fn new(start_epoch: &Epoch, num_of_epochs: u64) -> Self {
-        let start_slot = start_epoch.start_slot(T::slots_per_epoch()) + 1;
+    ///
+    /// NOTE: Removed the shift by one for deneb because otherwise the last batch before the blob
+    /// fork boundary will be of mixed type (all blocks and one last blockblob), and I don't want to
+    /// deal with this for now.
+    /// This means finalization might be slower in deneb
```

Comment:
I recall the purpose of this was that when we sync the last epoch/batch we downloaded could be finalized in fork-choice.  I think originally it was handy to debug as we got confirmation that we have successfully processed and finalized a whole epoch. 

After the fork, do we intend to shift this back? 


Off the top of my head, I can't see why current shift won't work, in principle we should start another batch even if there is only 1 slot to download and that should finalize the previous batch. I guess we have to wait to download and process the next batch before we see the finalization. I thought there was a more significant downside, but if thats it, seem alright?

I think potentially the other reason we went down this path was because it was designed to sync from genesis, and we always had the genesis block. So it made sense to shift by one to avoid re-downloading one block. 

If this new system works fine, then I think we probably should remove the comments above which explains the shift, just so its not confusing for the next person to look at what is going on.

Reply:
> After the fork, do we intend to shift this back?

Yea I think we can whenever we are able to stop supporting the capella -> deneb transition, so maybe in a couple forks. I'll leave the note in case we end up switchiing it back